scripts/install_linux_deps.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -e
+
+. /etc/os-release
+DISTRIB=$ID
+
+ARCH_DEPS="boost boost-libs expect jq autoconf shellcheck sqlite python-virtualenv"
+UBUNTU_DEPS="libboost-all-dev expect jq autoconf shellcheck sqlite3 python3-venv"
+
+if [ "${DISTRIB}" = "arch" ]; then
+    pacman -S --refresh --needed --noconfirm $ARCH_DEPS
```

Comment:
The idea is that this script is run with root privileges. The idea in the main file, I think, is that if sudo is not installed, the root is running the script. Otherwise, we are not able to install sudo anyway. But I'm actually not sure what's the use case. All distributions ship sudo I think.

Reply:
I think being upfront about root access is appropriate for a script like this. Taking it a step further and telling the user to run `sudo configure_dev.sh` rather than calling sudo inside the script would be even better.

Reply:
This script also installs go dependencies which should be done as a user. But we can make our own `--noconfirm` flag and depending on that pass or not pass `--noconfirm` to pacman or apt-get.

Reply:
I agree that the go dependencies should be installed as a user. Having these installed as root would cause havoc once the user try to use these as non-root and get `permission denied`.

Reply:
> The idea is that this script is run with root privileges. The idea in the main file, I think, is that if sudo is not installed, the root is running the script. Otherwise, we are not able to install sudo anyway. But I'm actually not sure what's the use case. All distributions ship sudo I think.

not all of them : https://unix.stackexchange.com/questions/48522/how-universal-is-sudo

Reply:
At the moment we only support ubuntu, maybe debian, right? Debian ships sudo starting version 9. The LTS for debian 8 has already ended.

Reply:
At the moment, we support the following linux distros: ubuntu 16+, Debian (raspbian), red-hat 7+

Reply:
Well, this script certainly doesn't support red-hat. In any case, is there anything wrong with the PR?

Reply:
If `apt-get install <ubuntu deps>` fails without sudo, then `apt-get install sudo` should also fail, right? I think the purpose of installing sudo was too eliminate code duplication. Instead of having
```
if no sudo
  apt-get install <long dependency list>
else
  sudo apt-get install <long dependency list>
```
it was
```
if no sudo
  install sudo
sudo apt-get install <long dependency list>
```

I changed the logic to the first and removed duplication by moving `<long dependency list>` to a constant.

But I agree that we should be careful. There is just no still supported distribution I'm aware of that uses both apt-get and doesn't have sudo. apt-get is debian specific and the last debian version without sudo (8) is already deprecated. So, we I think we can remove the 'no sudo' case, but it doesn't need to happen in this PR.

Reply:
I don't think apt-get run without root or sudo would have much effect anyway. The interesting thing here is the first check is if there's no sudo, install it, which implies you are root. Being root removes the need to sudo in the first place.

Perhaps a better solution is to replace the sudo with a test to see if you are root, and if so, run commands without sudo. That would preserve the ability to run as root, but also allow it if you have sudo privileges (and deny it if you don't).

Reply:
I honestly think that it's better to just have instructions about setting up a dev environment. I personally avoid using such scripts because they might have unintended consequences. In any case, this PR is just an incremental improvement and I tried to not change a lot.